src/main/java/dk/kb/datahandler/facade/DsDatahandlerFacade.java

```diff
@@ -118,100 +126,105 @@ public static String indexSolrDelta(String origin) throws IOException, SolrServe
-
-    	
-     	long lastModifiedForOrigin= getLastModifiedTimeForOrigin(originStatistics, oaiTargetDto.getDatasource());
+    public static Integer oaiIngestDelta(String oaiTargetName) throws Exception {   
+        OaiTargetDto oaiTargetDto = ServiceConfig.getOaiTargets().get(oaiTargetName);       
+        String lastHarvestTime = HarvestTimeUtil.loadLastHarvestTime(oaiTargetDto);
+
+        //Will be 1 interval for OAI targets that does not need to split into days
+        ArrayList<OaiFromUntilInterval>  intervals= HarvestTimeUtil.generateFromUntilInterval(oaiTargetDto, lastHarvestTime);
+        Integer totalHarvested = oaiIngestJobScheduler(oaiTargetName, intervals);                     
+        log.info("Delta ingest of target={} completed with records={}",oaiTargetName,totalHarvested);
+        return totalHarvested;    	    	
+    }
+    
+  
         
-        //register job
-        OaiJobCache.addNewJob(job);            
-
-        try {
-
-            int number= oaiIngest(job , null);        
-            OaiJobCache.finishJob(job, number,false);//No error
+   
 
-            //Delete old records in storage from before.
-            Integer numberDeleted = dsAPI.deleteRecordsForOrigin(oaiTargetDto.getDatasource(), 0L, lastModifiedForOrigin);
-            log.info("After full ingest for origin={}, deleted {} old records in storage",
-                    oaiTargetDto.getDatasource(), numberDeleted);
-            return number;
-        }
-        catch(Exception e) {
-            log.error("Oai full harvest did not complete succesfull:"+oaiTargetName);
-            OaiJobCache.finishJob(job, 0, true);//Error
-            job.setCompletedTime(System.currentTimeMillis());                                    
-            throw new Exception(e);
-        }
+    
+    /**
+     * This method has no specific code for the different OAI targets. Dateformats must be set correct for the target when calling this method. <br>
+     * The list of date-intervals must be ascending in time<br> 
+     * The date intervals will be harvested in same order as in list. After each interval harvest they persistent last harvesttime will be updated for that OAI target.
+     *  
+     * For each interval this method will start a new OAI job and call {@link #oaiIngestPerform()  oaiIngestPerform method}<br> 
+     *  
+     * @param oaiTargetName the name of the configured oai-target
+     * @param fromUntilList List of date intervals. When calling this method the date formats must be in format accepted by the target.
+     * 
+     * @return Total number of records harvest from all intervals. Records that are discarded will not be counted.
+     *      
+     */
+     protected static Integer oaiIngestJobScheduler(String oaiTargetName,ArrayList<OaiFromUntilInterval> fromUntilList) throws Exception {                
+         int totalNumber=0;
+                  
+         log.info("Starting jobs from number of FromUntilIntervals:"+fromUntilList.size());
```

Comment:
Please provide content here: the `oaiTargetName`.

Reply:
fixed